x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlRealm.java

```diff
@@ -744,25 +785,69 @@ private static Tuple<AbstractReloadingMetadataResolver, Supplier<EntityDescripto
         FileWatcher watcher = new FileWatcher(path);
         watcher.addListener(new FileListener(logger, resolver::refresh));
         watcherService.add(watcher, ResourceWatcherService.Frequency.MEDIUM);
-        return new Tuple<>(resolver, () -> resolveEntityDescriptor(resolver, entityId, path.toString()));
+        return new Tuple<>(resolver, () -> resolveEntityDescriptor(resolver, entityId, path.toString(), true));
     }
 
     private static EntityDescriptor resolveEntityDescriptor(
         AbstractReloadingMetadataResolver resolver,
         String entityId,
-        String sourceLocation
+        String sourceLocation,
+        boolean failOnError
     ) {
         try {
-            final EntityDescriptor descriptor = resolver.resolveSingle(new CriteriaSet(new EntityIdCriterion(entityId)));
+            final EntityDescriptor descriptor = resolveEntityDescriptorWithPossibleRefresh(resolver, entityId);
             if (descriptor == null) {
-                throw SamlUtils.samlException("Cannot find metadata for entity [{}] in [{}]", entityId, sourceLocation);
+                if (failOnError) {
+                    throw SamlUtils.samlException("Cannot find metadata for entity [{}] in [{}]", entityId, sourceLocation);
+                } else {
+                    logger.warn(
+                        "cannot load SAML metadata for [{}] from [{}]; SAML authentication for this realm will fail",
```

Comment:
Nit: might want to include realm name here. It can be derived by the entityId and url. But it is convenient for readers to just call it out.

Reply:
Unfortunately, we don't know the realm name here and it's tricky to resolve that.